package.json

```diff
@@ -17,7 +17,8 @@
 		"dev": "vite demo",
 		"dev:build": "vite build demo",
 		"build": "rimraf dist && tsc && tsc -p tsconfig.cjs.json && node tools/postbuild.mjs",
-		"prepublishOnly": "npm run build"
+		"prepublishOnly": "npm run build",
+		"prepare": "npm run build"
```

Comment:
This causes the `build` command to be run whenever a user installs our vite preset. Is this change necessary to allow JSX Import Source to be overwritten?

Reply:
According to the docs, I think it's only when the preset is installed through git? Which, of course, would require `build` be ran.

Reply:
I think it's not just that:

> - Runs on local `npm install` without any arguments

In either way this change doesn't seem necessary to allow JSX import source to be overwritten.

Reply:
> I think it's not just that:

`microbundle` has a prepare script, for example, and it doesn't run on install (nor could it). "local" means this project, as in, if you were to clone this repo and run `npm install`.

Not the best wording from NPM there, to be fair.

> In either way this change doesn't seem necessary to allow JSX import source to be overwritten.

Yeah totally agree there, and I generally think it's a super annoying life cycle script anyways.

Reply:
I stand corrected!

Reply:
I am guessing to keep the pull request pure, you want to remove the changes to package.json?

Reply:
That, but also as it stands, you haven't provided an explanation for the change. Why are we adding a new script?

While we can guess about your intentions (as we've done above), for right now, it's a change without any rhyme nor reason.

Reply:
To be fair, the `prepare` script isn't needed if the package is installed from npm. However, as I needed to overwrite the import source for a project, and did not want to publish to npm, I added the `prepare` script so that installing via git would be possible.